src/settings-ui/Settings.UI/Controls/ShortcutControl/ShortcutControl.xaml.cs

```diff
@@ -17,8 +17,7 @@ namespace Microsoft.PowerToys.Settings.UI.Controls
     public sealed partial class ShortcutControl : UserControl, IDisposable
     {
         private readonly UIntPtr ignoreKeyEventFlag = (UIntPtr)0x5555;
-        private bool _shiftKeyDownOnEntering;
-        private bool _shiftToggled;
+        private System.Collections.Generic.Dictionary<VirtualKey, bool> _modifierKeysOnEntering = new System.Collections.Generic.Dictionary<VirtualKey, bool>();
```

Comment:
We're not using `bool` anywhere, so it should be a `HashSet`?

Reply:
Yes, only the keys of the Dictionary are used, so it makes more sense to use HashSet instead, I will correct this, thanks😀